Source/Eto.Gtk/Forms/Controls/TreeGridViewHandler.cs

```diff
@@ -166,6 +166,44 @@ public override void AttachEvent(string id)
 			}
 		}
 
+		[GLib.ConnectBefore]
+		protected override void OnTreeButtonPress(object sender, Gtk.ButtonPressEventArgs e)
+		{
+			// If clicked mouse button is the primary button
+			if (e.Event.Button != 1)
+				return;
+
+			Gtk.TreePath path;
+			Gtk.TreeViewColumn clickedColumn;
+
+			// Get path and column from mouse position
+			Tree.GetPathAtPos((int)e.Event.X, (int)e.Event.Y, out path, out clickedColumn);
+
+			if (path == null || clickedColumn == null)
+				return;
+
+			int rowIndex = 0;
+			if (path.Indices.Length > 0)
+			{
+				for (int i = 0; i < path.Depth; i++)
+					rowIndex += path.Indices[i] + 1;
+
+				rowIndex--;
+			}
```

Comment:
This part makes sure that the rowIndex is the right value.
I tested it in the test application and the row index is the same as the row index from the SelectionChanged event.